### PR TITLE
Hotfixes, just fresh from the oven!

### PR DESCRIPTION
**Pull request by cygnusv**

### CODE REVIEW

File: nucypher/blockchain/eth/interfaces.py
Comment (derekpierre) [inline nucypher/blockchain/eth/interfaces.py:149]: I guess this commit never made it in because #2369  wasn't merged... @KPrasch cherry-picked it

File: nucypher/network/nodes.py
Comment (derekpierre) [inline nucypher/network/nodes.py:1362]: This will help with the monitor - the other problematic line was https://github.com/nucypher/nucypher/blob/ad6b6bda51a548f1b5592a3a7050a7c92d8bbb10/nucypher/network/nodes.py#L353.

Comment (derekpierre) [reply nucypher/network/nodes.py:1362]: My current workaround for the monitor - https://github.com/derekpierre/nucypher/commit/ad2e2d0524d90d368c5fb90f71c1c1bf3b46b10a

Comment (cygnusv) [reply nucypher/network/nodes.py:1362]: That's line is tricky, because I know that it doesn't fail in other contexts (e.g., when it's a file-based node storage). Not sure how to proceed here until #2356 is merged. What about an ugly try block?

File: nucypher/blockchain/eth/interfaces.py
Comment (cygnusv) [reply nucypher/blockchain/eth/interfaces.py:149]: Yeah, and I actually changed it a bit more since then.

File: nucypher/network/nodes.py
Comment (derekpierre) [reply nucypher/network/nodes.py:1362]: I'm good with the ugly for now - with a todo to fix properly later. Just gets us moving before a more comprehensive solution.

Comment (cygnusv) [reply nucypher/network/nodes.py:1362]: What do you think of this for line 353?
```
            try:  # Workaround until #2356 is fixed
                node_domain = node.domain.decode('utf-8')
            except:
                node_domain = node.serving_domain
```

Comment (derekpierre) [reply nucypher/network/nodes.py:1365]: Looks good. We can be specific about the `AttributeError` - but looks good to me anyway.

Comment (cygnusv) [reply nucypher/network/nodes.py:1365]: I'm not sure if it's an attribute error or a BytestringSplitter error. In any case, it's ugly af and we need to fix #2356 soon 😅

**Review by derekpierre — APPROVED** [review]
🎸

File: nucypher/blockchain/eth/interfaces.py
Comment (fjarri) [inline nucypher/blockchain/eth/interfaces.py:150]: Don't think `else` is necessary here. You're returning from the exception branch anyway.

**Comment by vepkenez** [comment]
I will not approve unless this PR has a fix for this:
```
	  File "/usr/local/lib/python3.7/site-packages/nucypher/network/nodes.py", line 1370, in abridged_node_details
	    known = self.known_nodes_details()
	  File "/usr/local/lib/python3.7/site-packages/nucypher/network/nodes.py", line 1334, in known_nodes_details
	    abridged_nodes[checksum_address] = self.node_details(node=node)
	  File "/usr/local/lib/python3.7/site-packages/nucypher/network/nodes.py", line 1362, in node_details
	    "domain": node.learning_domain,
	AttributeError: 'Ursula' object has no attribute 'learning_domain'
```

**Review by vepkenez — APPROVED** [review]

**Review by tuxxy — APPROVED** [review]

File: nucypher/network/nodes.py
Comment (KPrasch) [reply nucypher/network/nodes.py:1365]: In a follow-up PR we can collapse `learning_domain` and `serving_domain` into `domain`.

**Review by KPrasch — APPROVED** [review]

**Pull request merged**
